e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,184 @@
+	Resource: "nodehealthchecks",
+}
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
+
+	// +kubebuilder:scaffold:scheme
+
+	// get the client or die
+	getConfig, err := config.GetConfig()
+	if err != nil {
+		Fail(fmt.Sprintf("Couldn't get kubeconfig %v", err))
+	}
+	clientSet, err = kubernetes.NewForConfig(getConfig)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(clientSet).NotTo(BeNil())
+
+	dynamicClient, err = dynamic.NewForConfig(getConfig)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(dynamicClient).NotTo(BeNil())
+
+	debug()
+
+	Expect(createPoisonPillConfig()).To(Succeed())
+	Expect(createPoisonPillResources()).To(Succeed())
+	Expect(createNHCResources()).To(Succeed())
+}, 10)
+
+func createPoisonPillConfig() error {
+	obj := unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind":       "PoisonPillConfig",
+			"apiVersion": "poison-pill.medik8s.io/v1alpha1",
+			"metadata": map[string]interface{}{
+				"name": "poison-pill-config",
+			},
+			"spec": map[string]interface{}{},
+		},
+	}
+
+	ppConfig, err := dynamicClient.
+		Resource(poisonPillConfigGVR).
+		Namespace("poison-pill").
+		Create(context.Background(), &obj, metav1.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+
+	t, ok, err := unstructured.NestedInt64(ppConfig.UnstructuredContent(), "spec", "safeTimeToAssumeNodeRebootedSeconds")
+	if !ok || err != nil {
+		return errors.Wrap(err, "Failed to extract poison pill config")
+	}
+	safeToAssumeNodeRebootTimeoutInSeconds = int(t)
+
+	return nil
+}
+
+func createNHCResources() error {
+	maxUnhealthy := intstr.Parse("49%")
+	nhc := v1alpha1.NodeHealthCheck{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "NodeHealthCheck",
+			APIVersion: v1alpha1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: v1alpha1.NodeHealthCheckSpec{
+			Selector: metav1.LabelSelector{},
+			UnhealthyConditions: []v1alpha1.UnhealthyCondition{
+				{
+					Type:     v1.NodeReady,
+					Status:   v1.ConditionUnknown,
+					Duration: metav1.Duration{Duration: time.Second * 20},
+				},
+			},
+			MaxUnhealthy: &maxUnhealthy,
+			RemediationTemplate: &v1.ObjectReference{
+				Kind:       "PoisonPillRemediationTemplate",
+				APIVersion: "poison-pill.medik8s.io/v1alpha1",
+				Name:       "poison-pill-template",
+				Namespace:  "default",
+			},
+		},
+	}
+	toUnstructured, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&nhc)
+	fmt.Println(toUnstructured)
+	Expect(err).NotTo(HaveOccurred())
+
+	_, err = dynamicClient.
+		Resource(nhcGVR).
+		Create(
+			context.Background(),
+			&unstructured.Unstructured{Object: toUnstructured},
+			metav1.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+
+	return nil
+}
+
+func debug() {
+	version, _ := clientSet.ServerVersion()
+	fmt.Fprint(GinkgoWriter, version)
+}
+
+func createPoisonPillResources() error {
```

Comment:
maybe rename to `createPoisonPillTemplate()`?

Reply:
done